compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/AwtWindow.desktop.kt

```diff
@@ -112,4 +136,491 @@ fun <T : Window> AwtWindow(
             showJob.cancel()
         }
     }
-}
\ No newline at end of file
+}
+
+/**
+ * Similar to the corresponding [Window] function, but additionally allows configuring the
+ * underlying AWT window before it has been made displayable by providing an [init] block.
+ *
+ * This is useful to:
+ * - Set window properties which cannot be changed after it has been made displayable, such as
+ *   [java.awt.Window.setType].
+ * - Adding listeners for events that can occur when the window becomes displayable/visible.
+ *
+ * IMPORTANT: this function should not be used to set properties which can be changed after the
+ * window has been made displayable. Doing so can cause your code to stop working in the future if
+ * a parameter that controls this property is added to this function.
+ * For example, if you set the window's minimum size in [init] and later a `minimumSize` parameter
+ * is added to this function, it will override your setting of the minimum size in [init].
+ *
+ * To set these kinds of properties, use this pattern instead:
+ * ```
+ * Window( ... ) {
+ *     // Window content here
+ *     LaunchedEffect(window) {
+ *         // Configure window here
+ *     }
+ * }
+ * ```
+ *
+ * @see Window
+ */
+@ExperimentalComposeUiApi
+@Composable
+fun SwingFrame(
```

Comment:
```suggestion
fun Window(
```

We need to be consistent with the current names, even if they aren't good.

The current name for creating `ComposeWindow` is also `Window`:
![image](https://github.com/user-attachments/assets/4f34b313-7a78-4c8a-b30b-88735d6d7a3b)

I.e., at the momement there are 2 `Window` - one tries to hide `ComposeWindow` as implementation details, one is an additional overload, specifically for `ComposeWindow`. In this PR we add one more overload for it.

Reply:
Overall, consistency is more important than the right names.

If we want the right names, we also need to rename `Window(create=` to `ComposableJFrame`, and `ComposeWindow` to `ComposeJFrame`, with deprecation of the old names. We can do that, but we should do that in a separate PR.

Reply:
> As an existing example, we have AwtWindow...

It is a mess, but the new function is related to the function I posted in the screenshot, not to `AwtWindow`